addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java

```diff
@@ -0,0 +1,235 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.kodi.internal.protocol;
+
+import java.net.URI;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * KodiClientSocket implements the low level communication to kodi through websocket. Usually this communication is done
+ * through port 9090
+ *
+ * @author Paul Frank
+ *
+ */
+public class KodiClientSocket {
+    private static final Logger logger = LoggerFactory.getLogger(KodiClientSocket.class);
+
+    private static final ExecutorService threadpool = Executors.newFixedThreadPool(3);
```

Comment:
You should not create thread pools this way, but rather use the existing `scheduler` in `KodiHandler` - you can pass it as a parameter to the constructor of KodiConnection/KodiClientSocket. You do not seem to use these threads for any long-running tasks, so it should be fine to use the existing pool.